pkg/api/store/cluster/cluster_store.go

```diff
@@ -50,6 +51,29 @@ func (r *Store) Create(apiContext *types.APIContext, schema *types.Schema, data
 		return nil, httperror.NewFieldAPIError(httperror.InvalidOption, "enableNetworkPolicy", err.Error())
 	}
 
+	if rawEKSConfig := data[managementv3.ClusterFieldAmazonElasticContainerServiceConfig]; rawEKSConfig != nil {
+		raw, err := json.Marshal(rawEKSConfig)
+		if err != nil {
+			return nil, fmt.Errorf("error marshalling eks config: %v", err)
+		}
+
+		eksConfig := &managementv3.AmazonElasticContainerServiceConfig{}
+		err = json.Unmarshal(raw, eksConfig)
+		if err != nil {
+			return nil, fmt.Errorf("error unmarshalling eks config: %v", err)
+		}
+
+		if data[managementv3.ClusterFieldAnnotations] == nil {
+			// This seems like it should be a map[string]string{} but supplying that causes annotations to blow up
+			// spectacularly
+			data[managementv3.ClusterFieldAnnotations] = make(map[string]interface{})
+		}
+
+		if annotations, ok := data[managementv3.ClusterFieldAnnotations].(map[string]interface{}); ok {
+			annotations[clusterstatus.TemporaryCredentialsAnnotationKey] = strconv.FormatBool(eksConfig.SessionToken != "")
+		}
+	}
+
```

Comment:
There is a library in norman norman/types/values/values.go to help retrieving and setting nested map. So i think the code can be simplified as:
```
sessionToken := values.GetValue(data, managementv3.ClusterFieldAmazonElasticContainerServiceConfig, SessionTokenKey)
annotation := values.GetValue(data, managementv3.ClusterFieldAnnotations)
m := toMap(annotaton)
values.PutValue(data, managementv3.ClusterFieldAnnotations, m)
```

Reply:
Incorporated this change into lines 53-60.